cmd/ci-secret-bootstrap/main.go

```diff
@@ -457,6 +469,142 @@ func writeSecrets(secretsMap map[string][]*coreapi.Secret, w io.Writer) error {
 	return nil
 }
 
+type comparable struct {
+	fields      sets.String
+	attachments sets.String
+	hasPassword bool
+}
+
+func costructBWItemsByName(items []bitwarden.Item) map[string]*comparable {
+	itemsByName := make(map[string]bitwarden.Item)
+
+	for _, item := range items {
+		itemsByName[item.Name] = item
+	}
+
+	bwComparableItemsByName := make(map[string]*comparable)
+	for name, item := range itemsByName {
+		comparableItem := &comparable{}
+
+		if len(item.Fields) > 0 {
+			fields := sets.NewString()
+			for _, item := range item.Fields {
+				fields.Insert(item.Name)
+			}
+
+			comparableItem.fields = fields
+		}
+
+		if len(item.Attachments) > 0 {
+			attachments := sets.NewString()
+			for _, attachment := range item.Attachments {
+				attachments.Insert(attachment.FileName)
+			}
+
+			comparableItem.attachments = attachments
+		}
+
+		if item.Login != nil {
+			comparableItem.hasPassword = true
+		}
+
+		bwComparableItemsByName[name] = comparableItem
+	}
+
+	return bwComparableItemsByName
+}
+
+func costructConfigItemsByName(config secretbootstrap.Config) map[string]*comparable {
+	cfgComparableItemsByName := make(map[string]*comparable)
+
+	for _, cfg := range config.Secrets {
+		for _, bwContext := range cfg.From {
+			if bwContext.BWItem != "" {
+				item := &comparable{
```

Comment:
NIT: init `item` only when needed.

```suggestion
				item, ok := cfgComparableItemsByName[bwContext.BWItem]
				if !ok {
					item = &comparable{
						fields:      sets.NewString(),
						attachments: sets.NewString(),
					}
					cfgComparableItemsByName[bwContext.BWItem] = item
				}
```